sdk/servicebus/service-bus/samples/typescript/src/advanced/sessionRoundRobin.ts

```diff
@@ -0,0 +1,146 @@
+/*
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  Licensed under the MIT Licence.
+
+  This sample demonstrates how you can continually read through all the available
+  sessions in a Service Bus queue or subscription.
+
+  Run the sendMessages sample with some session ids before running this sample.
+*/
+
+import {
+  ServiceBusClient,
+  delay,
+  ReceivedMessageWithLock,
+  SessionReceiver,
+  MessagingError
+} from "@azure/service-bus";
+import * as dotenv from "dotenv";
+import { env } from "process";
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+
+dotenv.config();
+
+const serviceBusConnectionString =
+  env["SERVICEBUS_CONNECTION_STRING"] || "<service bus connection string not in environment>";
+
+// NOTE: this sample uses a queue but would also work a session enabled subscription.
+const queueName = env["QUEUE_NAME_WITH_SESSIONS"] || "<queue name not in environment>";
+
+const maxSessionsToProcessSimultaneously = 8;
+const sessionIdleTimeoutMs = 3 * 1000;
+const delayWhenNoSessionsAvailableMs = 5 * 1000;
+
+// this can be used control when the round-robin processing will terminate.
+const abortController = new AbortController();
+
+// called just before we start processing the first message of a session
+async function processInitialize(sessionId: string) {
+  console.log(`[${sessionId}] will start processing...`);
+}
+
+// called when we get a message for a session
+async function processMessage(msg: ReceivedMessageWithLock) {
+  console.log(`[${msg.sessionId}] received message with body ${msg.body}`);
+}
+
+// called if we get an error
+async function processError(err: Error, sessionId?: string) {
+  console.log(`[${sessionId}] had error ${err.message}`);
+}
+
+// Called if we are closing a session
+// `reason` will be:
+// * 'error' if we are closing because of an error(the error will be delivered
+//   to `processError` above)
+// * 'idle_timeout' if we `sessionIdleTimeoutMs` milliseconds pass without
+//   any messages being received (ie, session can be considered empty).
+//
+async function processClose(reason: "error" | "idle_timeout", sessionId: string) {
+  if (reason === "error") {
+    console.log(`[${sessionId}] was closed because of error`);
+  } else if (reason === "idle_timeout") {
+    console.log(`[${sessionId}] was closed - no more messages within idle timeout`);
+  }
```

Comment:
Since you have captured the two reasons in the comments above the function here, we can skip the same in code. Or skip the comment and keep the code

```js
console.log(`[${sessionId}] was closed because of ${reason}`)
```

Reply:
Done!